src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -94,7 +91,8 @@ internal PackageSourceMoniker SelectedSource
             IVsShell4 vsShell,
             INuGetUILogger uiLogger = null)
         {
-            _uiDispatcher = Dispatcher.CurrentDispatcher;
+            VSThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
why VSThreadHelper instead of normal ThreadHelper?

Reply:
I wondered the same thing. `VSThreadHelper` _is_ `MS.VS.Shell.ThreadHelper`. Check the `using` directives at the top of the file.